### ISSUE TITLE
May Update: Problem with run Browser at Debugging

### ISSUE THREAD

**Issue by DanielGoehler** [Microsoft/AL#2409]
On May Update (13.0.22267.0 DE), which I create from "Daily Builds - Developing for a future release of Dynamics 365 Business Central" as Azure VM I experienced the following repeatedly behavior:

1. I can publish and install a simple app ([Simple Project.zip](https://github.com/Microsoft/AL/files/2047961/Simple.Project.zip)) without "startupObjectId" in lunch.json, but there is an error because chrome is missing.
**lunch.json:**
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "al",
            "request": "launch",
            "name": "Your own server",
            "server": "https://bcsandbox.westeurope.cloudapp.azure.com",
            "serverInstance": "NAV",
            "tenant": "default",
            "authentication": "UserPassword"
        }
    ]
}
```
**Output:**
```
Microsoft (R) AL Compiler version 0.17.16611
Copyright (C) Microsoft Corporation. All rights reserved

Success: The package is created.
[2018-05-29 11:34:57.87] Publishing package to tenant 'default'
[2018-05-29 11:34:57.87] Sending request to https://bcsandbox.westeurope.cloudapp.azure.com:7049/NAV/dev/apps?tenant=default&SchemaUpdateMode=synchronize
[2018-05-29 11:34:58.93] Success: The package 'DAGO_Simple Sample_1.0.0.0.app' has been published to the server.
[2018-05-29 11:34:58.93] Sending request to https://bcsandbox.westeurope.cloudapp.azure.com:7049/NAV/dev/webendpoint?tenant=default
The system cannot find the file chrome.
```
2. I can publish and install a simple app with "startupObjectId" in lunch.json, but there is an internal error at the end.
**lunch.json:**
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "al",
            "request": "launch",
            "name": "Your own server",
            "server": "https://bcsandbox.westeurope.cloudapp.azure.com",
            "serverInstance": "NAV",
            "tenant": "default",
            "authentication": "UserPassword",
            "startupObjectId": 22,
            "startupObjectType": "Page"
        }
    ]
}
```
**Output:**
```
[2018-05-29 11:37:53.18] Publishing package to tenant 'default'
[2018-05-29 11:37:53.18] Sending request to https://bcsandbox.westeurope.cloudapp.azure.com:7049/NAV/dev/apps?tenant=default&SchemaUpdateMode=synchronize
[2018-05-29 11:37:54.14] Success: The package 'DAGO_Simple Sample_1.0.0.0.app' has been published to the server.
[2018-05-29 11:37:54.24] Error: An internal error has occurred
```

**Comment by thpeder** [comment]
Hi @DanielGoehler,
1) It looks like your default browser is misconfigured. In a terminal try running: start https://github.com/Microsoft/AL/issues/2409
Since that is just what we are doing in the extension, you should get a similar error.

2) Could you take a look at the eventviewer of your server.
We can see from telemetry that the debugger fails to start so could you check in the Microsoft.Dynamics.Nav.Server.exe.config file that NetFx40_LegacySecurityPolicy is disabled.

**Comment by thpeder** [comment]
You can open a powershell session in the container like so: ```docker exec -ti MyDockerContainerName powershell```

**Comment by thpeder** [comment]
We found the problem with why the debugger doesn't work and the fix should be available sometime next week.

**Comment by salgiza** [comment]
We are having the same problem with the "12.2.22932.23097 (ES)" docker release. When we try to debug any of our extensions, the result is always the same:

```
[2018-07-11 11:10:49.70] Publishing package to tenant 'default'
[2018-07-11 11:10:49.71] Sending request to https://bcdocker.westeurope.cloudapp.azure.com:7049/nav/dev/apps?tenant=default&SchemaUpdateMode=synchronize
[2018-07-11 11:10:51.01] Success: The package 'XXX_1.0.0.0.app' has been published to the server.
[2018-07-11 11:10:51.11] Error: An internal error has occurred
```

We are using version 0.16.19969 of the AL extension. I've checked that NetFx40_LegacySecurityPolicy is disabled in the docker, and there is nothing in the docker event log that points to any problems.

Worst of all, this is a new VM that we created after our previous VM stopped working (see #2839 ).
Previously to that, though, debugging worked without problems (that machine was "12.1.22475.22720 (ES)"), so it might seem like the problem is due to a change done between both releases.

**Comment by thpeder** [comment]
@salgiza does the event viewer contain some more information?

**Comment by salgiza** [comment]
I haven't seen anything when running

`Get-NavContainerEventLog navserver`

(Only a lot of info messages, but none seem relevant to the problem, and there are no warnings or errors).

In the log of the machine that is running VS Code, there are no warning or error messages in the Application Windows Event Log, either.

Is there any other place I should check for logs?

**Comment by DanielGoehler** [comment]
@thpeder Which version should I try?

**Comment by salgiza** [comment]
Just a quick update: we finally deployed a new docker specifying the old version that worked ("12.1.22475.22720 (ES)") and the error stopped. Before that, we also tried with an US docker ("12.2.22932.23097 (US)) that gave us the same error.

So, in our case:

12.1.22475.22720 (ES) => Works.
12.2.22932.23097 (ES) => Error: An internal error has occurred
12.2.22932.23097 (US) => Error: An internal error has occurred

All of them using version 0.16.19969 of the AL extension.

We can't stay in the old version forever, though, so if there is anything that we can do to help you find the cause of the bug, @thpeder , please say so.

**Comment by thpeder** [comment]
Hi @salgiza,
I think you are hitting another issue here (#2867), namely that you can't launch the debugger, but the extensions does get published, so if you use: `AL: Publish without debugger (Ctrl + F5)` then it should open the browser as expected.

**Comment by salgiza** [comment]
Hi @thpeder , you are right, the problem only happens when running the debugger; the extension is always successfully deployed in the server, in either case. 

P.S. Unless I read @DanielGoehler description of this ticket incorrectly, though, this is the same bug that happened to him, when he selected an startupObjectId.

**Comment by DanielGoehler** [comment]
Hi @thpeder, with 12.2.22932 DE I still have this Problem with F5. Ctrl + F5 works.

**Comment by dbezzant** [comment]
So is this still an issue?  If so, how do I go back to an earlier build?  Not being able to debug while learning all this new process is very frustrating.

**Comment by salgiza** [comment]
@dbezzant You can specify the version you want when downloading dockers/deploying them in Azure: 
https://blogs.msdn.microsoft.com/freddyk/2018/04/16/which-docker-image-is-the-right-for-you/

We are still using old docker versions for development, due to this bug (or #2867 , which seems to be the same).

**Comment by dbezzant** [comment]
@salgiza will I run into any issues developing in an older build and deploying to a customers Business Central?

**Comment by salgiza** [comment]
@dbezzant “minor” releases (in this case, anything that is 12.x) should always be backwards compatible, so you shouldn’t have any problems until Business Central customers are updated with the fall update (13.x), in October/November.

**Comment by StanislawStempin** [comment]
This problem has been fixed on version 12.3 as well as in the July developer preview (13.0.X)
